### PR TITLE
[tooltip] add programatic show/hide capability

### PR DESCRIPTION
**Pull request by scottjason**
Added a isVisible prop (If you don't pass anything it just works as it does now (sets to false on render)).

cc: @dcocchia @nbwar @spencerschack

### CODE REVIEW

**Comment by scottjason** [comment]
It gets a little confusing to sort of have this both ways (have two different variables controlling the visibility). So right now w the pr as is, it's being instructed by the `isVisible` prop when `true` and then by the `state.show` when `false`.

**Comment by spencerschack** [comment]
I would add a test for the `isVisible` prop and instead of adding another instance of the tooltip in the example, I would add a set of radio buttons that control `isVisible`

**Comment by scottjason** [comment]
@spencerschack ok.

Also, what do you think about the fact that this is being controlled both by the `isVisible` prop which is coming from outside the component as well as the internal state of the component `isShow`?

**Comment by spencerschack** [comment]
I think that's what Dom and I were originally thinking this would work, not sure how else it would be done.

File: src/components/Tooltip/Tooltip.js [hunk 1]
```diff
@@ -70,11 +70,13 @@ class Tooltip extends PureComponent {
...
 
   render() {
+
     const {
       children,
       placement,
       size,
-      snacksStyle
+      snacksStyle,
+      isVisible,
```
Comment (dcocchia) [inline src/components/Tooltip/Tooltip.js]: be sure to add this to the `propTypes`

Comment (scottjason) [reply src/components/Tooltip/Tooltip.js]: 👍

**Comment by scottjason** [comment]
@spencerschack Ok, yeah I think it should be fine. My other thought was to pass in the `isVisible` prop from the outside (or not) and have that define `state.show` (if and only if it's defined, not falsey).

But. there's problems w that i think because you'd have to set `this.state.show` in the render method.

**Comment by scottjason** [comment]
@spencerschack Ok, now sharing the same tooltip (although I really don't think this makes any sense...this button is setup already to handle things using the internal state of the Tooltip component and now we're kind of blending in the other).

**Comment by scottjason** [comment]
@spencerschack Ok I updated the PR and got rid of the extra example all together and now all of the examples are using the `isVisible` propr.

What i was originally trying to add as an example was showing how to use say an actual radio button as the target (like what we need to do in low stock) instead of having to click the radio button and then the toggle button.

**Pull request merged**